### PR TITLE
Feature/3702 export data

### PR DESCRIPTION
**Pull request by cw-cvp**

### CODE REVIEW

File: src/components/export/ExportTab/ExportTab.js [hunk 1]
```diff
@@ -71,13 +78,47 @@ const ExportTab = ({
...
+      const mpName = mpRes.data["name"];
+      const date = new Date();
+      const year = date.getUTCFullYear();
+      const month = date.getUTCMonth() + 1;
+      const day = date.getUTCDate();
+      const fullDateString = `${month}-${day}-${year}`;
+      const facRes = await getFacilityById(facId)
+      const facName = facRes.data["facilityName"];
+      exportFileName = `MP Export - ${facName}, ${mpName} (${fullDateString}).json`;
+      download(JSON.stringify(mpRes.data, null, "\t"), exportFileName);
```
Comment (subrataroyCVP) [inline src/components/export/ExportTab/ExportTab.js]: You could have use `exportHandler` function from `src/components/HeaderInfo/HeaderInfo.js` instead for duplicating same code.

**Review by subrataroyCVP — COMMENTED** [review]
Otherwise LGTM.

**Review by Qb10 — APPROVED** [review]

**Review by subrataroyCVP — CHANGES_REQUESTED** [review]
Its not exporting selected rows when export button is clicked. Its exporting all the rows available in that data table even 1 or 2 selected..

image

**Comment by subrataroyCVP** [comment]
When MonitorPlan and QA Cert both selected its only downloading `MonitorPlan.json`

File: src/components/export/ExportTab/ExportTab.js [hunk 2]
```diff
@@ -71,13 +75,36 @@ const ExportTab = ({
...
+      setPreviewOptions({ beginDate, endDate });
     }
   };
+
+  const exportClickHandler = async () => {
+    let exportFileName
+    // export monitoring plan
+    if (dataTypes.find((e) => e.name === "monitoring-plan").checked) {
+      exportMonitoringPlanDownload(selectedConfig.id)
+      return;
```
Comment (subrataroyCVP) [inline src/components/export/ExportTab/ExportTab.js]:
```suggestion
```

File: src/components/export/ExportTab/ExportTab.js [hunk 3]
```diff
@@ -71,13 +75,36 @@ const ExportTab = ({
...
+    }
+    // export if preview was clicked
+    exportFileName = `Export - ${facility}.json`;
+    const exportJson = exportState.qaTestSummaryRows;
+    download(JSON.stringify(exportJson, null, "\t"), exportFileName);
```
Comment (subrataroyCVP) [inline src/components/export/ExportTab/ExportTab.js]:
```suggestion
if (dataTypes.find((e) => e.name === "qa-and-certification").checked) {
      exportFileName = `QA & Certification | Export - ${facility}.json`;
      const exportJson = {
        orisCode: orisCode,
        testSummaryData: exportState.qaTestSummaryRows
      }
      download(JSON.stringify(exportJson, null, "\t"), exportFileName);
    }
```

File: src/components/export/ExportTab/ExportTablesContainer.js [hunk 4]
```diff
@@ -68,7 +57,7 @@ const ExportTablesContainer = ({
       {
         ...exportState,
         selectedIds,
-        qaTestSummaryRows: rows,
+        qaTestSummaryRows: qaTestSummaryData,
```
Comment (subrataroyCVP) [inline src/components/export/ExportTab/ExportTablesContainer.js:60]:
```suggestion
        qaTestSummaryRows: selectedRows,
```

**Review by subrataroyCVP — CHANGES_REQUESTED** [review]
Suggested the code changes

File: src/components/export/ExportTab/ExportTab.js (hunk 1 above)
Comment (cw-cvp) [reply src/components/export/ExportTab/ExportTab.js]: moved export monitoring plan into exported function

**Comment by subrataroyCVP** [comment]
LGTM.

**Pull request merged**